aten/src/ATen/native/cuda/Nonzero.cu

```diff
@@ -184,9 +193,15 @@ void nonzero_cuda_out_impl(const Tensor& self, Tensor& out) {
   auto num_nonzeros = allocator.allocate(sizeof(int) * num_chunks);
   for (int64_t idx = 0; idx < num_chunks; idx++) {
     int64_t remaining = std::min(chunk_size, self.numel() - idx * chunk_size);
+#if CUB_V3_PLUS()
+    thrust::transform_iterator<NonZeroOp<scalar_t>, const scalar_t*> itr(
+        self_.const_data_ptr<scalar_t>() + idx * chunk_size,
+        NonZeroOp<scalar_t>());
+#else
     cub::TransformInputIterator<bool, NonZeroOp<scalar_t>, const scalar_t*> itr(
         self_.const_data_ptr<scalar_t>() + idx * chunk_size,
         NonZeroOp<scalar_t>());
+#endif
```

Comment:
Nitpick: this could also leverage `thrust::make_transform_iterator`
```suggestion
    thrust::transform_iterator<NonZeroOp<scalar_t>, const scalar_t*> itr(
        self_.const_data_ptr<scalar_t>() + idx * chunk_size,
        NonZeroOp<scalar_t>());
```